### PR TITLE
Update Private Repo Instructions

### PR DESCRIPTION
**Pull request by stuart23**
The instructions in the current doc do not work. We have had two customers use these instructions and not being able to get it to work ([9203](https://astronomer.zendesk.com/agent/tickets/9203) & [9443](https://astronomer.zendesk.com/agent/tickets/9443)).

These changes have been tested with StyleSeat and have been confirmed to work.

### CODE REVIEW

File: software/customize-image.md [hunk 1]
```diff
@@ -264,7 +264,13 @@ To install Python packages from a private GitHub repository on Astronomer Softwa
 - A [Software project](create-project.md).
 - Custom Python packages that are [installable via pip](https://packaging.python.org/en/latest/tutorials/packaging-projects/).
 - A private GitHub repository for each of your custom Python packages.
-- A [GitHub SSH Key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) authorized to access your private GitHub repositories.
+- A [GitHub SSH Private Key](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent) authorized to access your private GitHub repositories.
+
+:::warning
+
+If your organization enforces SAML Single Sign On (SSO), you must first authorize your key to allow it to be used by following [this guide](https://docs.github.com/en/enterprise-cloud@latest/authentication/authenticating-with-saml-single-sign-on/authorizing-an-ssh-key-for-use-with-saml-single-sign-on).
```
Comment (paolaperaza) [inline software/customize-image.md:271]:
```suggestion
If your organization enforces SAML single sign-on (SSO), you must first authorize your key to be used with that authentication method. For instructions, see [GitHub documentation](https://docs.github.com/en/enterprise-cloud@latest/authentication/authenticating-with-saml-single-sign-on/authorizing-an-ssh-key-for-use-with-saml-single-sign-on).
```

File: software/customize-image.md [hunk 2]
```diff
@@ -336,19 +339,24 @@ This example assumes that the name of each of your Python packages is identical
 
   :::tip
 
-  If you don't want keys in this file to be pushed back up to your GitHub repository, consider adding this file to `.gitignore`.
+  Older versions of the Astronomer Certified image contain different versions of
+  Python. To find the python version in an image, use
+  `docker run quay.io/astronomer/ap-airflow:2.2.5 python --version`. The version
+  numbers in the *Copy requirements directory* command above may need to be
+  changed if a different version of Python is being used.
```
Comment (paolaperaza) [inline software/customize-image.md:346]:
```suggestion
This example `Dockerfile.build` assumes Python 3.9, but some versions of Astronomer Certified may be based on a different version of Python. If your image is based on a version of Python that is not 3.9, replace `python 3.9` in the **COPY** commands listed under the `## Copy requirements directory` section of your `Dockerfile.build` with the correct Python version.
  
  To identify the Python version in your AC image, run:
  
  `
  docker run quay.io/astronomer/ap-airflow:<astronomer-certified-version> python --version`
  
  Make sure to replace `<astronomer-certified-version>` with your own. 
```

File: software/customize-image.md [hunk 3]
```diff
@@ -336,19 +339,24 @@ This example assumes that the name of each of your Python packages is identical
...
 
   :::
 
   :::info
 
-  If your repository is hosted somewhere other than GitHub, replace the location of your SSH key in the `ssh-keyscan` command.
+  If your repository is hosted somewhere other than GitHub, replace the domain
+  in the `ssh-keyscan` command with the domain where the package is hosted.
```
Comment (paolaperaza) [inline software/customize-image.md:353]:
```suggestion
  If your repository is hosted somewhere other than GitHub, replace the domain in the `ssh-keyscan` command with the domain where the package is hosted.
```

**Review by paolaperaza — APPROVED** [review]
LGTM @stuart23 ! I saw those support tickets 😣 

Made some copy changes. Wanna take a look?

File: astro/develop-project.md [hunk 4]
```diff
@@ -452,46 +455,47 @@ This example assumes that the name of each of your Python packages is identical
 
   :::tip
 
-  If you don't want keys in this file to be pushed back up to your GitHub repository, consider adding this file to `.gitignore`.
+  This example `Dockerfile.build` assumes Python 3.9, but some versions of Astro Runtime may be based on a different version of Python. If your image is based on a version of Python that is not 3.9, replace `python 3.9` in the **COPY** commands listed under the `## Copy requirements directory` section of your `Dockerfile.build` with the correct Python version.
+  
+  To identify the Python version in your Astro Runtime image, run:
+  
+  `
+  docker run quay.io/astronomer/astro-runtime:<runtime-version>-base python --version`
```
Comment (paolaperaza) [inline astro/develop-project.md:463]:
```suggestion
     ```
     docker run quay.io/astronomer/astro-runtime:<runtime-version>-base python --version
     ```
```

File: software/customize-image.md [hunk 5]
```diff
@@ -336,19 +339,26 @@ This example assumes that the name of each of your Python packages is identical
 
   :::tip
 
-  If you don't want keys in this file to be pushed back up to your GitHub repository, consider adding this file to `.gitignore`.
+This example `Dockerfile.build` assumes Python 3.9, but some versions of Astronomer Certified may be based on a different version of Python. If your image is based on a version of Python that is not 3.9, replace `python 3.9` in the **COPY** commands listed under the `## Copy requirements directory` section of your `Dockerfile.build` with the correct Python version.
+  
+  To identify the Python version in your AC image, run:
+  
+  `
+  docker run quay.io/astronomer/ap-airflow:<astronomer-certified-version> python --version`
```
Comment (paolaperaza) [inline software/customize-image.md:347]:
```suggestion
     ```
     docker run quay.io/astronomer/ap-airflow:<astronomer-certified-version> python --version
     ```
```

**Comment by stuart23** [comment]
Also fixes https://github.com/astronomer/docs/issues/641

**Pull request merged**